app/controllers/posts_controller.rb

```diff
@@ -168,10 +160,12 @@ def raw_email
   end
 
   def short_link
-    post = Post.find(params[:post_id].to_i)
+    post = Post.find_by(id: params[:post_id].to_i)
+    raise Discourse::NotFound unless post
+
     # Stuff the user in the request object, because that's what IncomingLink wants
     if params[:user_id]
-      user = User.find(params[:user_id].to_i)
+      user = User.find_by(id: params[:user_id].to_i)
```

Comment:
I know this is not your change, but why do we cast `.to_i` here and on line 163?

```
[4] pry(main)> User.find_by(id: -1).username
  User Load (2.3ms)  SELECT "users".* FROM "users" WHERE "users"."id" = -1 LIMIT 1
=> "system"
[5] pry(main)> User.find_by(id: "-1").username
  User Load (3.2ms)  SELECT "users".* FROM "users" WHERE "users"."id" = -1 LIMIT 1
=> "system"
```

Reply:
To be extra correct, I guess 😛 I see we're doing that in at least 25 places in core.